features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplay.kt

```diff
@@ -30,21 +30,26 @@ object SessionReplay {
         sessionReplayConfiguration: SessionReplayConfiguration,
         sdkCore: SdkCore = Datadog.getInstance()
     ) {
-        val sessionReplayFeature = SessionReplayFeature(
-            sdkCore = sdkCore as FeatureSdkCore,
-            customEndpointUrl = sessionReplayConfiguration.customEndpointUrl,
-            privacy = sessionReplayConfiguration.privacy,
-            imagePrivacy = sessionReplayConfiguration.imagePrivacy,
-            touchPrivacy = sessionReplayConfiguration.touchPrivacy,
-            textAndInputPrivacy = sessionReplayConfiguration.textAndInputPrivacy,
-            customMappers = sessionReplayConfiguration.customMappers,
-            customOptionSelectorDetectors = sessionReplayConfiguration.customOptionSelectorDetectors,
-            sampleRate = sessionReplayConfiguration.sampleRate,
-            startRecordingImmediately = sessionReplayConfiguration.startRecordingImmediately,
-            dynamicOptimizationEnabled = sessionReplayConfiguration.dynamicOptimizationEnabled
-        )
-
-        sdkCore.registerFeature(sessionReplayFeature)
+        val featureSdkCore = sdkCore as FeatureSdkCore
+        sessionReplayConfiguration.systemRequirementConfiguration.runIfMeetRequirements(
+            sdkCore = sdkCore,
+            featureSdkCore.internalLogger
+        ) {
+            val sessionReplayFeature = SessionReplayFeature(
+                sdkCore = featureSdkCore,
+                customEndpointUrl = sessionReplayConfiguration.customEndpointUrl,
+                privacy = sessionReplayConfiguration.privacy,
+                imagePrivacy = sessionReplayConfiguration.imagePrivacy,
+                touchPrivacy = sessionReplayConfiguration.touchPrivacy,
+                textAndInputPrivacy = sessionReplayConfiguration.textAndInputPrivacy,
+                customMappers = sessionReplayConfiguration.customMappers,
+                customOptionSelectorDetectors = sessionReplayConfiguration.customOptionSelectorDetectors,
+                sampleRate = sessionReplayConfiguration.sampleRate,
+                startRecordingImmediately = sessionReplayConfiguration.startRecordingImmediately,
+                dynamicOptimizationEnabled = sessionReplayConfiguration.dynamicOptimizationEnabled
+            )
+            sdkCore.registerFeature(sessionReplayFeature)
```

Comment:
What was the decision in the end regarding the thread that this will be called on?

Reply:
The decision I take is that we register the feature in main thread in the end no matter where the user call `enable` function.